pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -60,6 +60,12 @@ func (r *pullthroughBlobStore) remoteStat(ctx context.Context, dgst digest.Diges
 		return distribution.Descriptor{}, err
 	}
 
+	r.pullFromInsecureRegistries = false
```

Comment:
I agree as well. I'm just saying that default value shouldn't be decided at such a low level.

By this change, `r.pullFromInsecureRegistries` is rendered totally useless. If it's no longer useful for pullthrough tests then please remove it and use local variable instead.